### DISCUSSION TITLE
Pi4 moved running Buster image to Bullseye to get SSD running, now Zigbee2MQTT wont start.

### DISCUSSION THREAD

**Discussion by acorupe** [Koenkk/zigbee2mqtt#18001] [Devices]
Hi I have been using Zigbee2MQTT for several months on my Pi4B with Buster, I have a Sonoff Co-ordinator and its been working really well added several devices, using in conjunction with OpenHab 3.4.

I decided to add a USB SDD boot drive and an Argon One case. After many attempts to hack the boot files on the old OS Buster, I decided a fresh install is the way to go so went with Bullseye. Got the SSD running right away with this approach, however everything else needs to get re-installed.

With the new build on Bullseye I am getting: 
     Error while starting zigbee-herdsman
     Exiting...
     Zigbee2MQTT:error: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
     
     I tried copying the /opt/zigbee2mqtt/data directory complete from the old image, unfortunately does not make a difference
     
     Checked that the tty port is correct in the configuration.yaml file
     and verified with  test -w /dev/ttyACM0 && echo success || echo failure
         and
    ls -l ls -l /dev/ttyACM0 is crw-rw---- 1 root dialout 166, 0 Jun 11 19:15 /dev/ttyACM0 
              
    Checked and set the group permissions on the device file and added my userid to the important groups
    
    I stopped modemmanager and HCIUart, still no go. 
    
    I do have existing zigbee devices that are paired, could this be the problem, PanID, Channel  or ?.
    
    If I go back to my old image and restart Zigbee2MQTT starts up fine. Bullseye looks like the issue but what?.
    
    Any ideas or suggestions appreciated.
    
    Allan

**Comment by acorupe** [comment]
More information on the issue,    did a DEBUG=zigbee-herdsman* npm start

Zigbee2MQTT:debug 2023-06-12 13:03:37: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":true},"backupPath":"/opt/zigbee2mqtt/data/coordinator_backup.json","databaseBackupPath":"/opt/zigbee2mqtt/data/database.db.backup","databasePath":"/opt/zigbee2mqtt/data/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":6754},"serialPort":{"path":"/dev/ttyACM0"}}'
  zigbee-herdsman:controller:log Starting with options '{"network":{"networkKeyDistribute":false,"networkKey":[1,3,5,7,9,11,13,15,0,2,4,6,8,10,12,13],"panID":6754,"extendedPanID":[221,221,221,221,221,221,221,221],"channelList":[11]},"serialPort":{"path":"/dev/ttyACM0"},"databasePath":"/opt/zigbee2mqtt/data/database.db","databaseBackupPath":"/opt/zigbee2mqtt/data/database.db.backup","backupPath":"/opt/zigbee2mqtt/data/coordinator_backup.json","adapter":{"disableLED":true,"concurrent":null,"delay":null}}' +0ms
  zigbee-herdsman:adapter:zStack:znp:log Opening SerialPort with {"path":"/dev/ttyACM0","baudRate":115200,"rtscts":false,"autoOpen":false} +0ms
  zigbee-herdsman:adapter:zStack:znp:log Serialport opened +8ms
  zigbee-herdsman:adapter:zStack:znp:SREQ --> SYS - ping - {"capabilities":1} +0ms
  zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,0,33,1,32] +0ms
  zigbee-herdsman:adapter:zStack:znp:log Writing CC2530/CC2531 skip bootloader payload +259ms
  zigbee-herdsman:adapter:zStack:unpi:writer --> buffer [239] +255ms
  zigbee-herdsman:adapter:zStack:znp:SREQ --> SYS - ping - {"capabilities":1} +1s
  zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,0,33,1,32] +1s
  zigbee-herdsman:adapter:zStack:znp:log Skip bootloader for CC2652/CC1352 +1s
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [17] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [17] +1ms
  zigbee-herdsman:adapter:zStack:znp:SREQ --> SYS - ping - {"capabilities":1} +710ms
  zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,0,33,1,32] +710ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [26,193,2,2,155,123,126] +997ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [17,26,193,2,2,155,123,126] +0ms
  zigbee-herdsman:adapter:zStack:znp:SREQ --> SYS - ping - {"capabilities":1} +6s
  zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,0,33,1,32] +6s
  zigbee-herdsman:adapter:zStack:znp:SREQ --> SYS - ping - {"capabilities":1} +6s
  zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,0,33,1,32] +6s
Zigbee2MQTT:error 2023-06-12 13:03:59: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2023-06-12 13:03:59: Failed to start zigbee
Zigbee2MQTT:error 2023-06-12 13:03:59: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2023-06-12 13:03:59: Exiting...
Zigbee2MQTT:error 2023-06-12 13:03:59: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.start (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:103:27)
    at Controller.start (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/opt/zigbee2mqtt/lib/zigbee.ts:58:27)
    at Controller.start (/opt/zigbee2mqtt/lib/controller.ts:101:27)
    at start (/opt/zigbee2mqtt/index.js:107:5)
allan@pi4:/opt/zigbee2mqtt$ ls -l /dev/ttyA*
crwxrwxrwx 1 root dialout 166, 0 Jun 12 13:03 /dev/ttyACM0
allan@pi4:/opt/zigbee2mqtt$

**Comment by Shred99** [comment]
I guess the big question is: are you sure that /dev/ACM0 is the correct port?

Try doing an "lsusb" to list all USB devices

"udevadm info --query=all --name=/dev/ttyACM0" could be revealing too.

From your post, it looks like you've read the documentation, so I assume you have added the user running Z2M to the groups uucp , tty and dialout.

**Comment by acorupe** [comment]
Thank you for your reply. i really appreciate the response,

The port /dev/ttyACM0 was confirmed. Per the docs, I also added the user to
the uucp, tty and dialout groups, I also removed modem manager and ran,
sudo systemctl disable hciurt.

I finally discovered the issue yesterday.

Turns out the Sonoff dongle I was sent is the ZBDongle-E which is in
experimental status.

I  added to the the   *serial:*
                                      *adapter:ezsp*

After which Zigbee2mqtt Heardsman starts up fine. While it's strange that
it starts ok with my original installation with the SD card OS
(Buster) *without
the adapter: ezsp* line in configuration file. Seems I got lucky there.

my new configuration.yaml file with Debian Bullseye

advanced:

  log_level: debug

homeassistant: false

permit_join: true

mqtt:

  base_topic: zigbee2mqtt

  server: mqtt://127.0.0.1:1883

serial:

  port: /dev/ttyACM0

  disable_led: true

  adapter: ezsp

frontend:

  port: 9090

  availability_timeout: 60

experimental:

  output: attribute

devices:

  '0x00158d0009438687':

    friendly_name: back_patio_door

    description: Aqara Back Door Sensor

Thank You so much !

Best Regards,

Allan Corupe
cell (602)568-8407

**Reply by Shred99** [reply]
Thanks for coming back and sharing the solution.  Hopefully it might save someone else from having the same problem in the future.
